packages/rollup-core/src/app/data/data-service.ts

```diff
@@ -248,38 +248,37 @@ export class DefaultDataService implements DataService {
       batchNumber: gethSubmissionNumber,
       timestamp,
       blockNumber,
-      batches: [
-        res.map((row: Row, indexWithinSubmission: number) => {
-          const tx: RollupTransaction = {
-            indexWithinSubmission,
-            l1TxHash: row['l1_tx_hash'],
-            l1TxIndex: row['l1_tx_index'],
-            l1TxLogIndex: row['l1_tx_log_index'],
-            target: row['target'],
-            calldata: row['calldata'], // TODO: may have to format Buffer => string
-            l1Timestamp: row['block_timestamp'],
-            l1BlockNumber: row['block_number'],
-            queueOrigin: row['queue_origin'],
-          }
-
-          if (!!row['sender']) {
-            tx.sender = row['sender']
-          }
-          if (!!row['l1MessageSender']) {
-            tx.l1MessageSender = row['l1_message_sender']
-          }
-          if (!!row['gas_limit']) {
-            tx.gasLimit = row['gas_limit']
-          }
-          if (!!row['nonce']) {
-            tx.nonce = row['nonce']
-          }
-          if (!!row['signature']) {
-            tx.nonce = row['signature']
-          }
-          return tx
-        }),
-      ],
+      rollupTransactions: res.map((row: Row, indexWithinSubmission: number) => {
+        const tx: RollupTransaction = {
+          l1RollupTxId: parseInt(row['id'], 10),
```

Comment:
Ah so adding the ID was basically trivial because we already had one assigned by the DB--that right?

Reply:
Yep. To be clear, the ID we're using is the auto-generated database unique ID (primary key). So it's just a matter of slapping it on the data structure.